### PR TITLE
Remove goog vec

### PR DESCRIPTION
**Pull request by nicholas-l**
Branch to try to change to glMatrix, currently very much a WIP

- [x] identity
- [x] translate
- [x] scale
- [x] createNumber
- [x] getElement
- [x] createNumberIdentity
- [x] invert
- [x] setFromArray
- [x] rotateZ
- [x] multMat
- [x] makeTranslate/makeRotateZ/makeScale
- [x] multVec4

### CODE REVIEW

**Comment by nicholas-l** [comment]
All tests are going to fail because there is two ```@fileoverview```s in the file meaning the ```@suppress``` gets ignored.

**Comment by nicholas-l** [comment]
Ready for review. Larger build size but maybe with ES6ify of glMatrix that @bjornharrtell is d[oing](https://github.com/toji/gl-matrix/pull/209), this could reduce.

**Comment by ahocevar** [comment]
Thanks for this huge effort @nicholas-l. Out of interest: I see that npmjs.com has many sub-packages of gl-matrix. Wouldn't it make sense to use only those needed, instead of including the whole thing? Also, is there a reason for not using the module from npmjs.com, but your branch instead?

**Comment by nicholas-l** [comment]
The reason for using my branch is that it fails when there is two fileoverviews (one from the package and one from the headers the externs script adds). I changed to my own branch in the start but can see if it works with the npm package. Also I think the separate npm packages are out of date (could be wrong :)). Eventually I see using the ES6 glmatrix when it comes out and it won't matter as we can import the modules we need.

**Comment by bjornharrtell** [comment]
I was motivated to do https://github.com/toji/gl-matrix/pull/209 for just that reason, to avoid having to split up a library into several packages. I'm a bit allergic to the package-as-module that the NPM registry has become. But it needs (alot?) more work both on glMatrix side and OpenLayers to be useful.

File: externs/gl-matrix.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+var SIMD = {};
```
Comment (fredj) [inline externs/gl-matrix.js]: not used?

**Comment by nicholas-l** [comment]
I get an error with the current package on NPM but I have already submitted a fix (and a subsequent one) to toji, and it has already been merged. After these fixes, I still get errors (473) like below:
```
ERR! compile /home/nicholas/Development/ol3/build/ol.ext/glmatrix.js:6490: ERROR - inconsistent return type
ERR! compile found   : boolean
ERR! compile required: (Boolean|null)
ERR! compile     return (Math.abs(a0 - b0) <= glMatrix.EPSILON*Math.max(1.0, Math.abs(a0), Math.abs(b0)) &&
```
which I think it is due to two ```@fileoverview```s and the suppresses not being applied.

File: externs/gl-matrix.js (hunk 1 above)
Comment (nicholas-l) [inline externs/gl-matrix.js]: Not used but the compiler complains about undefined global variable.

**Comment by tschaub** [comment]
I haven't dug into it yet, but I was hoping to first do a bit of an audit into our vec/mat use.  I'm curious, for example, to see if we are doing a lot of mat4 operations where we could be doing mat3 operations.  Are there places where we need mat4 currently?

**Comment by nicholas-l** [comment]
Not sure but I assumed that the Mat4 operations were due to eventual view/3D/tilting.

**Comment by probins** [comment]
@nicholas-l now that #5335 has been merged, could you rename the typedef `ol.vec.Mat4.Number` to say `ol.Mat4Number` and move it to `typedefs.js`. This will bring it into line with the other typedefs. Thanks.

**Comment by nicholas-l** [comment]
@probins sorry about the radio silence! Yes, will rebase and change typedef

**Pull request closed without merging**